apps/web/src/components/Publication/FullPublication.tsx

```diff
@@ -4,26 +4,34 @@ import getAppName from '@lib/getAppName';
 import dayjs from 'dayjs';
 import relativeTime from 'dayjs/plugin/relativeTime';
 import type { Publication } from 'lens';
-import type { FC } from 'react';
+import type { FC, Ref } from 'react';
+import { useLayoutEffect, useRef } from 'react';
 
 import PublicationActions from './Actions';
 import PublicationMenu from './Actions/Menu';
 import HiddenPublication from './HiddenPublication';
 import PublicationBody from './PublicationBody';
 import PublicationStats from './PublicationStats';
+import ThreadBody from './ThreadBody';
 import PublicationType from './Type';
 
 dayjs.extend(relativeTime);
 
 interface Props {
   publication: Publication;
+  postContainerRef?: Ref<HTMLDivElement>;
 }
 
-const FullPublication: FC<Props> = ({ publication }) => {
+const FullPublication: FC<Props> = ({ publication, postContainerRef }) => {
+  const threadRef = useRef<HTMLDivElement>(null);
+
   const isMirror = publication.__typename === 'Mirror';
   const profile = isMirror ? publication?.mirrorOf?.profile : publication?.profile;
   const timestamp = isMirror ? publication?.mirrorOf?.createdAt : publication?.createdAt;
 
+  const commentOn = publication.__typename === 'Comment' ? (publication?.commentOn as any) : null;
+  const mainPost = commentOn && commentOn?.__typename !== 'Mirror' ? commentOn?.mainPost : null;
```

Comment:
Do we need `commentOn &&` condition? cuz the second one has optional chaining in it